### PR TITLE
add default deny rule for wasmplugin & fix default network allow-all

### PR DESCRIPTION
**Pull request by MaYuan-02**
**Please provide a description of this PR:**

This PR aims to fix this issue #53279
* Add default deny rule for network/http wasmplugin when pull wasm image fails.
* fix default allow-all network rbac rule, cause `StatPrefix` in network rbac filter is required.

### CODE REVIEW

File: pkg/wasm/convert_test.go [hunk 1]
```diff
@@ -287,14 +288,15 @@ func TestWasmConvert(t *testing.T) {
 			wantErr: false,
 		},
 		{
+			// 需要修正
```
Comment (ramaraochavali) [inline pkg/wasm/convert_test.go:291]: can you please remove this?

File: pkg/wasm/convert.go [hunk 2]
```diff
@@ -38,23 +41,104 @@ var (
 	allowHTTPTypedConfig = &anypb.Any{
 		TypeUrl: "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC",
 	}
-	allowNetworkTypedConfig = &anypb.Any{
-		TypeUrl: "type.googleapis.com/envoy.extensions.filters.network.rbac.v3.RBAC",
-	}
+	denyHTTPTypedConfig, _ = anypb.New(&httprbac.RBAC{
+		Rules: &rbacv3.RBAC{
+			Action: rbacv3.RBAC_DENY,
+			Policies: map[string]*rbacv3.Policy{
```
Comment (ramaraochavali) [inline pkg/wasm/convert.go:47]: Do we need to specify these policies?

File: pkg/wasm/convert.go [hunk 3]
```diff
@@ -38,23 +41,104 @@ var (
...
+					},
+				},
+			},
+		},
+	})
+
+	allowNetworkTypeConfig, _ = anypb.New(&networkrbac.RBAC{
+		Rules: &rbacv3.RBAC{
+			Action: rbacv3.RBAC_ALLOW,
+			Policies: map[string]*rbacv3.Policy{
```
Comment (ramaraochavali) [inline pkg/wasm/convert.go:71]: Why is this changed?

Comment (MaYuan-02) [reply pkg/wasm/convert.go:71]: Before the modification, this was an empty configuration. Envoy requires that stat_prefix cannot be empty, so it rejects this configuration. However, the httpfilter does not have this requirement.

After testing, I found that only this type of configuration can achieve the purpose of allow-all. Therefore, I also made changes here.

File: pkg/wasm/convert.go (hunk 2 above)
Comment (MaYuan-02) [reply pkg/wasm/convert.go:47]: Do you have better options?
This is also a point I was uncertain about when implementing this here. Currently, this implementation leads to the return of 'access denied by rbac', and using fault injection is also a good option.

Comment (MaYuan-02) [reply pkg/wasm/convert.go:47]: Do you have better options?
This is also a point I was uncertain about when implementing this here. Currently, this implementation leads to the return of 'access denied by rbac', and using fault injection is also a good option.

Comment (hzxuzhonghu) [reply pkg/wasm/convert.go:47]: cannot you leave it empty

Comment (ramaraochavali) [reply pkg/wasm/convert.go:47]: I was thiniking leaving it empty with DENY Action should help as well

File: pkg/wasm/convert_test.go (hunk 1 above)
Comment (MaYuan-02) [reply pkg/wasm/convert_test.go]: resolved

File: pkg/wasm/convert.go (hunk 3 above)
Comment (MaYuan-02) [reply pkg/wasm/convert.go]: resolved

**Review by ramaraochavali — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

File: releasenotes/notes/53279.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+# issue is a list of GitHub issues resolved in this note.
+# If issue is not in the current repo, specify its full URL instead.
+issue:
+  - https://github.com/istio/istio/issues/53279
+  - 23624
+
+releaseNotes:
+- |
+  **Fixed** For a WasmPlugin of type FAIL_CLOSE, if the wasm image fetch fails, a DENY-ALL RBAC filter will be used.
```
Comment (dhawton) [inline releasenotes/notes/53279.yaml:13]:
```suggestion
  **Fixed** an issue where if a wasm image fetch fails, an allow all RBAC filter is used. Now if `failStrategy` is set to `FAIL_CLOSE`, a DENY-ALL RBAC filter will be used.
```

**Comment by MaYuan-02** [comment]
@zirain Hello, this PR requires a designated Code Owner for review, but I'm not sure who to approach. Could you help me find the right person? Thank you!
> istio/wg-policies-and-telemetry-maintainers was requested for review as a code owner

**Review by zirain — APPROVED** [review]

**Comment by zirain** [comment]
@istio/wg-test-and-release-maintainers

**Comment by MaYuan-02** [comment]
@dhawton I would greatly appreciate it if you could review this PR again.😊

**Review by kfaseela — APPROVED** [review]

**Comment by kfaseela** [comment]
why are the tests very flaky? @zirain

**Comment by kfaseela** [comment]
@MaYuan-02 : Will you be interested in taking a look at this issue? https://github.com/istio/istio/issues/51288

**Comment by howardjohn** [comment]
Yeah this broke TestBadWasmRemoteLoad
